### PR TITLE
chore(settings): Stake inference beta config

### PR DESCRIPTION
**Pull request by danielSanchezQ**
## 1. What does this PR implement?
This adds the learning rate (*B*) as a deployment config instead of a constant.

## 2. Does the code have enough context to be clearly understood?

Should be simple

## 3. Who are the specification authors and who is accountable for this PR?

@davidrusu @danielSanchezQ @ntn-x2 

## 4. Is the specification accurate and complete?

Yes

## 5. Does the implementation introduce changes in the specification?

No

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [x] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [x] 2. Description added.
* [x] 3. Context and links to Specification document(s) added.
* [x] 4. Main contact(s) (developers and specification authors) added
* [x] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 6. Link PR to a specific milestone.

### CODE REVIEW

File: ledger/src/cryptarchia/stake.rs [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
 
-/// Current learning rate as per [especification](https://nomos-tech.notion.site/Total-Stake-Inference-22d261aa09df8051a454caa46ec54b34), this is not configurable.
-pub const LEARNING_RATE: u64 = 1;
 pub const PRECISION: u64 = 1000;
 
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[derive(Copy, Clone)]
 pub struct StakeInference {
-    learning_rate: u64,
+    learning_rate: f64,
```
Comment (ntn-x2) [inline ledger/src/cryptarchia/stake.rs:8]: If this cannot be zero, we have dedicated `F64` types to enforce that. At the very least we should use the `NonNegative` one, but even the `greater-than-zero` one if it makes sense.

File: nodes/node/binary/src/config/cryptarchia/deployment.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ use serde::{Deserialize, Serialize};
 pub struct Settings {
     pub epoch_config: EpochConfig,
     pub security_param: NonZeroU32,
+    pub learning_rate: f64,
```
Comment (ntn-x2) [inline nodes/node/binary/src/config/cryptarchia/deployment.rs]: Same as above: do we want to enforce additional constraints, so we fail at deserialization time?

File: nodes/node/binary/src/config/deployment/devnet.rs [hunk 3]
```diff
@@ -34,6 +34,7 @@ cryptarchia:
     epoch_period_nonce_buffer: 3
     epoch_period_nonce_stabilization: 4
   security_param: 20
+  learning_rate: 0.1
```
Comment (ntn-x2) [inline nodes/node/binary/src/config/deployment/devnet.rs:37]: Is it `0.1` or `1.0`?

Comment (davidrusu) [reply nodes/node/binary/src/config/deployment/devnet.rs:37]: we settled on 0.1 for devnet

File: ledger/src/cryptarchia/stake.rs (hunk 1 above)
Comment (davidrusu) [reply ledger/src/cryptarchia/stake.rs:8]: IMO, if we want to add more constraints to this, don't do it in this PR, lets log a techdebt issue as to not delay the release for devnet today.

Comment (danielSanchezQ) [reply ledger/src/cryptarchia/stake.rs:8]: Yes, I was searching for those, but only found the ones for integers. I'll check again. Thanks.

File: nodes/node/binary/src/config/deployment/devnet.rs (hunk 3 above)
Comment (danielSanchezQ) [reply nodes/node/binary/src/config/deployment/devnet.rs:37]: its `0.1` for devnet iirc

File: ledger/src/cryptarchia/stake.rs (hunk 1 above)
Comment (danielSanchezQ) [reply ledger/src/cryptarchia/stake.rs:8]: > IMO, if we want to add more constraints to this, don't do it in this PR, lets log a techdebt issue as to not delay the release for devnet today.

I left it like that here, ideally we should have a NonZeroF64. Inputs from config uses the NonNegativeF64 now, so it should be fine.
